### PR TITLE
GH-317: [Java] add recipe for vector schema root appender cases

### PR DESCRIPTION
**Pull request by davisusanibar**
To solve #317

### CODE REVIEW

File: java/source/data.rst [hunk 1]
```diff
@@ -23,6 +23,85 @@ Recipes related to compare, filtering or transforming data.
...
+    import org.apache.arrow.dataset.scanner.Scanner;
+    import org.apache.arrow.dataset.source.Dataset;
+    import org.apache.arrow.dataset.source.DatasetFactory;
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.VectorSchemaRoot;
+    import org.apache.arrow.vector.ipc.ArrowReader;
+    import org.apache.arrow.vector.util.VectorSchemaRootAppender;
+
+    VectorSchemaRoot appendVectorSchemaRootAsOne(BufferAllocator allocator) {
```
Comment (lidavidm) [inline java/source/data.rst:51]: I think it would be clearer if we just constructed the vectors in-memory instead of having to pull in dataset for this

Comment (davisusanibar) [reply java/source/data.rst]: Changed

File: java/source/data.rst [hunk 2]
```diff
@@ -23,6 +23,69 @@ Recipes related to compare, filtering or transforming data.
...
 
+Append VectorSchemaRoots
+========================
+
+In some cases, VectorSchemaRoot needs to be modeled as a container. To accomplish
+this, you can use ``VectorSchemaRootAppender.append``. The following code reads a
+Parquet file with three row groups, gets the three vectors separately, and then
+appends the three vectors together:
+
+There will be another cases of integration w
```
Comment (lidavidm) [inline java/source/data.rst:34]: Cut off?

File: java/source/data.rst [hunk 3]
```diff
@@ -23,6 +23,69 @@ Recipes related to compare, filtering or transforming data.
 
 .. contents::
 
+Append VectorSchemaRoots
+========================
+
+In some cases, VectorSchemaRoot needs to be modeled as a container. To accomplish
+this, you can use ``VectorSchemaRootAppender.append``. The following code reads a
+Parquet file with three row groups, gets the three vectors separately, and then
+appends the three vectors together:
```
Comment (lidavidm) [inline java/source/data.rst:32]: Text needs updating

File: java/source/data.rst [hunk 4]
```diff
@@ -23,6 +23,69 @@ Recipes related to compare, filtering or transforming data.
...
+            appenderTwo.allocateNew(2);
+            appenderTwo.set(0, 34);
+            appenderTwo.set(1, 75);
+            appenderTwo.setValueCount(2);
+            try (
+                final VectorSchemaRoot rootOne = new VectorSchemaRoot(
+                        Collections.singletonList(appenderOne));
+                final VectorSchemaRoot rootTwo = new VectorSchemaRoot(
+                        Collections.singletonList(appenderTwo))
+            ) {
```
Comment (lidavidm) [inline java/source/data.rst:67]: Why so much nesting? This can be written more clearly.

**Review by lidavidm — CHANGES_REQUESTED** [review]

File: java/source/data.rst (hunk 4 above)
Comment (davisusanibar) [reply java/source/data.rst]: Updated

File: java/source/data.rst (hunk 2 above)
Comment (davisusanibar) [reply java/source/data.rst]: Deleted

File: java/source/data.rst [hunk 5]
```diff
@@ -23,6 +23,57 @@ Recipes related to compare, filtering or transforming data.
...
+        appenderTwo.set(0, 34);
+        appenderTwo.set(1, 75);
+        appenderTwo.setValueCount(2);
+        try (
+            final VectorSchemaRoot rootOne = new VectorSchemaRoot(
+                    Collections.singletonList(appenderOne));
+            final VectorSchemaRoot rootTwo = new VectorSchemaRoot(
+                    Collections.singletonList(appenderTwo))
+        ) {
+            if (result == null) {
```
Comment (lidavidm) [inline java/source/data.rst:59]: result is always null...

File: java/source/data.rst [hunk 6]
```diff
@@ -23,6 +23,57 @@ Recipes related to compare, filtering or transforming data.
...
+                result = VectorSchemaRoot.create(rootOne.getSchema(), allocator);
+                result.allocateNew();
+            }
+            VectorSchemaRootAppender.append(result, rootOne, rootTwo);
+            System.out.print(result.contentToTSVString());
+        }
+        result.close();
```
Comment (lidavidm) [inline java/source/data.rst:66]: try-with-resources

File: java/source/data.rst [hunk 7]
```diff
@@ -23,6 +23,57 @@ Recipes related to compare, filtering or transforming data.
...
+        appenderOne.allocateNew(2);
+        appenderOne.set(0, 100);
+        appenderOne.set(1, 20);
+        appenderOne.setValueCount(2);
+        appenderTwo.allocateNew(2);
+        appenderTwo.set(0, 34);
+        appenderTwo.set(1, 75);
+        appenderTwo.setValueCount(2);
+        try (
+            final VectorSchemaRoot rootOne = new VectorSchemaRoot(
```
Comment (lidavidm) [inline java/source/data.rst:54]: you can just use a schema and `VectorSchemaRoot.create` to get rid of the nesting here by initializing everything in the top-level try-with-resources

**Review by lidavidm — CHANGES_REQUESTED** [review]

File: java/source/data.rst (hunk 5 above)
Comment (davisusanibar) [reply java/source/data.rst]: Changed

File: java/source/data.rst (hunk 6 above)
Comment (davisusanibar) [reply java/source/data.rst]: Added

File: java/source/data.rst [hunk 8]
```diff
@@ -23,6 +23,58 @@ Recipes related to compare, filtering or transforming data.
 
 .. contents::
 
+Append VectorSchemaRoots
+========================
```
Comment (lidavidm) [inline java/source/data.rst:27]:
```suggestion
Concatenating VectorSchemaRoots
===============================
```

File: java/source/data.rst [hunk 9]
```diff
@@ -23,6 +23,58 @@ Recipes related to compare, filtering or transforming data.
...
+
+In some cases, VectorSchemaRoot needs to be modeled as a container. To accomplish
+this, you can use ``VectorSchemaRootAppender.append``. The following code gets
+the three vectors separately, and then appends the three vectors together:
```
Comment (lidavidm) [inline java/source/data.rst:31]:
```suggestion
this, you can use ``VectorSchemaRootAppender.append``. The following code 
creates two roots, then concatenates them together:
```

**Review by lidavidm — APPROVED** [review]

**Pull request merged**